lib/active_record/connection_adapters/fb/schema_statements.rb

```diff
@@ -79,33 +79,54 @@ def create_table(name, options = {}) # :nodoc:
 
           return if options[:sequence] == false || !needs_sequence
           create_sequence(options[:sequence] || default_sequence_name(name))
+          trg_sql = <<-END_SQL
+            CREATE TRIGGER N$#{name.upcase} FOR #{name.upcase}
+            ACTIVE BEFORE INSERT
+            AS
+            declare variable gen_val bigint;
+            BEGIN
+              if (new.ID is null) then
+                new.ID = next value for #{options[:sequence] || default_sequence_name(name)};
+              else begin
+                gen_val = gen_id(#{options[:sequence] || default_sequence_name(name)}, 0);
+                if (new.ID > gen_val) then
+                  gen_val = gen_id(#{options[:sequence] || default_sequence_name(name)}, new.ID - gen_val);
+              end
+            END
+          END_SQL
+          execute(trg_sql)
+          commit_db_transaction
         end
 
         def rename_table(name, new_name)
           fail ActiveRecordError, 'Firebird does not support renaming tables.'
         end
 
         def drop_table(name, options = {}) # :nodoc:
+          super
```

Comment:
If the table has a sequence, you can't drop it. You need to drop the sequence first.

The `super` should come after the sequence is dropped.